lib/experimental/fonts/font-library/class-wp-rest-font-faces-controller.php

```diff
@@ -367,9 +365,28 @@ public function create_item( $request ) {
 	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
 	 */
 	public function delete_item( $request ) {
+		$post = $this->get_post( $request['id'] );
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		$font_family = $this->get_parent_font_family_post( $request['font_family_id'] );
+		if ( is_wp_error( $font_family ) ) {
+			return $font_family;
+		}
+
+		if ( (int) $font_family->ID !== (int) $post->post_parent ) {
+			return new WP_Error(
+				'rest_font_face_parent_id_mismatch',
+				/* translators: %d: A post id. */
+				sprintf( __( 'The font face does not belong to the specified font family with id of "%d"', 'gutenberg' ), $font_family->ID ),
+				array( 'status' => 404 )
+			);
+		}
+
 		$force = isset( $request['force'] ) ? (bool) $request['force'] : false;
```

Comment:
Check that the parent font family exists and belongs to the font face before deleting the font face.